test/DisplayCervicalCancerMedicalHistory/cases/BiopsyResultsOnlyObservationHasProperCode.yml

```diff
@@ -0,0 +1,32 @@
+---
+name: Biopsy Results Only Observation Has Proper Code
+
+data:
+-
+  resourceType: Patient
+  name: Joanne Smith
+  gender: female
+  birthDate: 1980-01-01
+  extension:
+  -
+    url: http://hl7.org/fhir/us/core/StructureDefinition/us-core-birthsex
+    valueCode: F
+-
+  resourceType: DiagnosticReport
+  code: urn:oid:1.2.840.114350.1.13.284.3.7.5.737384.95#SURG01
```

Comment:
Open questions as I still don't fully understand the whole CQL logic yet.

1. The requirement is to "capture cervical biopsy test where the LOINC code to classify the test as a biopsy was in the Observation, as opposed to the Diagnostic Report".  If so, do we still need DiagnosticReport in test scenario? Looks like the test fails if I remove DiagnosticReport.
2. One of the DoDs are: "Check that the original patients mentioned by Carolann (above) are now displaying biopsy results." Should we test the dashboard if it shows the result `SNOMEDCT#285836003 CIN 1`, or test at CQL level is enough? I imagine setting up test on the dashboard is another work, set up sandbox with the simulated patient data, then run the rebuilt CQL and dashboard on top of that. Maybe it's not in the scope?

Reply:
Looked back the PR summary above, if the requirement is this "this expression now includes not only the DiagnosticReports that have a "Biopsy Report" LOINC code, but also the DiagnosticReports that reference an Observation with this DiagnosticReport", should the DiagnosticReport in test has reference to Observation?

Reply:
## Answers to open questions (comment 1):

### Question 1:

According to U.S. Core, as well as the data mapping we've done with our pilot partner, a lab test will be represented by a [US Core DiagnosticReport](https://build.fhir.org/ig/HL7/US-Core/StructureDefinition-us-core-diagnosticreport-lab.html). This resource groups and summarizes a lab test. The DiagnosticReport will then reference Observations represented by the [US Core Lab Result Observation Profile](https://build.fhir.org/ig/HL7/US-Core/StructureDefinition-us-core-observation-lab.html). These Observations represent individual lab tests and, and the values of those tests.
	
What we have found in practice with our pilot site is that the DiagnosticReport will always exist, and represent something like "cervical biopsy". Then there will be separate referenced Observations for things like biopsy result, specimen adequacy, etc.
	
Our CQL is currently set up so that the list of testing history is based upon the available DiagnosticReports. We then reach into Observations to look for test results. The test directories that start with "WithObs…" show how this is modeled when both a DiagnosticReport and Observation are present. But we have not found the presence of just an Observation to be a way that lab reports are represented.

### Question 2

I actually did some testing in the Dashboard when testing my branch. The easiest way I found to do this was to add a patient to https://github.com/ccsm-cds-tools/ccsm-cds-dashboard/blob/pilot/src/test/fhir/testData.js, and then run that patient through the Dashboard on http://localhost:3000/tests-fhir/. You can pretty easily generate a FHIR patient using the CQL Testing Framework. The test cases created in YAML are translated into a FHIR Bundle, which is then outputted in the `test_results` folder. For example, the patient bundle for the test case I created in this PR is located at `test/DisplayCervicalCancerMedicalHistory/test_results/DashboardLibrary_v1.0.0/bundles/Biopsy_Results_Only_Observation_Has_Proper_Code.json`. Let me know if you would like me to show you the results of that testing, or walk through how you could set up your own environment to test in this way.

Reply:
## Response to comment about referencing an Observation in the test (comment 2):

This is probably a good idea. This is the function that ends up getting called to determine what Observations are associated with a DiagnosticReport: https://github.com/ccsm-cds-tools/ccsm-cds-with-tests/blob/missing-histology-in-patient-history/cql/DashboardLibrary.cql#L869-L875. I associated the resources in the test I added by making the dates the same. But it would probably be worth adding a test where they are associated by reference, and one where they are associated by the `basedOn` field. I can add this.

Reply:
I added https://github.com/ccsm-cds-tools/ccsm-cds-with-tests/pull/148/commits/74a52f9544e0003bfd93f44f6334504079031166 to address your second comment in this thread.